src/CMakeLists.txt

```diff
@@ -637,12 +620,22 @@ if(${WITH_RADOSGW})
     ${CURL_LIBRARIES}
     ${EXPAT_LIBRARIES}
   )
-endif(${WITH_RADOSGW})
-if(${WITH_RBD})
+endif(WITH_RADOSGW)
+if(WITH_RBD)
   set(DENCODER_EXTRALIBS
       ${DENCODER_EXTRALIBS}
       rbd_replay_types)
-endif(${WITH_RBD})
+  if(LINUX)
+    list(APPEND dencoder_srcs
+        $<TARGET_OBJECTS:krbd_objs>)
+  endif(LINUX)
+endif(WITH_RBD)
+if(WITH_KVS)
```

Comment:
On 2-8-2016 09:58, Kefu Chai wrote:
> In src/CMakeLists.txt
> <https://github.com/ceph/ceph/pull/10417#discussion_r73109596>:
> 
>>    set(DENCODER_EXTRALIBS
>>        ${DENCODER_EXTRALIBS}
>>        rbd_replay_types)
>> -endif(${WITH_RBD})
>> +  if(LINUX)
>> +    list(APPEND dencoder_srcs
>> +        $<TARGET_OBJECTS:krbd_objs>)
>> +  endif(LINUX)
>> +endif(WITH_RBD)
>> +if(WITH_KVS)
> 
>     But the KeyValue store seems to use the file common/secret.c which
>     includes keyutils.h.
> 
> |secret.c| is used by |mount.ceph.c| and |krbd.cc| also. i am not sure
> how you come to this.


I just looked at the source of secret.c. :)
Perhaps it can be removed, but have not yet dared to do so.
KV exclusion for this also seems to work.